vj4/model/adaptor/setting.py

```diff
@@ -59,14 +63,22 @@
             desc='Choose the background image in your profile page.',
             image_class='user-profile-bg--thumbnail-{0}')]
 
-SETTINGS = PREFERENCE_SETTINGS + ACCOUNT_SETTINGS
+
+DOMAIN_USER_SETTINGS = [
+    Setting('setting_info_domain', 'domain_user_display_name', str,
+            name='Display Name')]
+DOMAIN_USER_SETTINGS_KEYS = set(s.key for s in DOMAIN_USER_SETTINGS)
+
+SETTINGS = PREFERENCE_SETTINGS + ACCOUNT_SETTINGS + DOMAIN_USER_SETTINGS
 SETTINGS_BY_KEY = collections.OrderedDict(zip((s.key for s in SETTINGS), SETTINGS))
 
 
 class SettingMixin(object):
   def get_setting(self, key):
     if self.has_priv(builtin.PRIV_USER_PROFILE) and key in self.user:
       return self.user[key]
+    if self.has_priv(builtin.PRIV_USER_PROFILE) and key.replace('domain_user_', '') in self.domain_user:
```

Comment:
Please do not use `str.replace` when the intent is to consume a prefix.

Please do not consume string prefix.